### DISCUSSION TITLE
Speeding up the vpn check tests

### DISCUSSION THREAD

**Discussion by mlopeza** [haugene/vpn-configs-contrib#62] [Ideas]
I just recently worked in a PR that made me test the configuration files I was working on, the whole testing process was a little bit difficult. There are some checks done but they are really slow as they use the main project. I think that ultimately using the tool would be nice as a final check, but to speed up pre-verifications I think we should work on something else.

I did some poking around this project: [nagios icinga openvp](https://github.com/liquidat/nagios-icinga-openvpn) and it looks it should be easy to do connection validations by parsing the configuration file and then using the code in this plugin to do the verification. This is of course dependent on the openvpn protocol and if something breaks in the future we might need to update it to support it (I think we should just support whatever the main project client supports).

I can work on this, just wanted to gather some thought around the idea.

**Comment by pkishino** [comment]
it would be awesome if there was some simple script that could even be triggered in the CI flow to validate and then push updates
Always happy to check PRs
